src/Domain/Common/Base/AggregateRoot.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System.ComponentModel.DataAnnotations.Schema;
+
+namespace SSW.CleanArchitecture.Domain.Common.Base;
+
+/// <summary>
+/// Cluster of objects treated as a single unit.
+/// Can contain entities, value objects, and other aggregates.
+/// Enforce business rules (i.e. invariants)
+/// Can be created externally.
+/// Can raise domain events.
+/// Represent a transactional boundary (i.e. all changes are saved or none are saved)
+/// </summary>
+public abstract class AggregateRoot<TId> : Entity<TId>, IAggregateRoot
+{
+    private readonly List<DomainEvent> _domainEvents = [];
+
+    [NotMapped]
+    public IReadOnlyList<DomainEvent> DomainEvents => _domainEvents.AsReadOnly();
+
+    public void AddDomainEvent(DomainEvent domainEvent) => _domainEvents.Add(domainEvent);
+
+    public void RemoveDomainEvent(DomainEvent domainEvent) => _domainEvents.Remove(domainEvent);
+
+    public void ClearDomainEvents() => _domainEvents.Clear();
+}
+
+public interface IAggregateRoot
+{
+}
```

Comment:
I still think the IAggregateRoot should contain the blueprint

```cs
public interface IAggregateRoot
{
    public IReadOnlyList<DomainEvent> DomainEvents { get; }

    public void AddDomainEvent(DomainEvent domainEvent);

    public void RemoveDomainEvent(DomainEvent domainEvent);

    public void ClearDomainEvents();
}
```

Reply:
Updated interface